### ISSUE TITLE
[bug] account stats get reset and show wrong followers, following, and post count 

### ISSUE THREAD

**Issue by Fastidious** [superseriousbusiness/gotosocial#3094]
### Describe the bug with a clear and concise description of what the bug is.

I am seeing a wrong count on followers, following, and posts. Running snapshot `b65bc`. 

### What's your GoToSocial Version?

3.5.3+0.16.1-SNAPSHOT+git-aeb65bc

### GoToSocial Arch

amd64, snapshot b65bc

### What happened?

I upgraded to snapshot `b65bc`. 

### What you expected to happen?

I expected followers, following, and post counts to remain about the same. I am following over 100 accounts, GtS shows 2. It also shows only 1 follower, and 7 posts. 

### How to reproduce it?

I figure that by running the same snapshot would be reproduced, but not sure.

**Comment by tsmethurst** [comment]
Could you be more specific? Are you seeing this via the API or via the web view?

**Comment by Fastidious** [comment]
I am seeing this on both, the API, and the web view.

**Comment by NyaaaWhatsUpDoc** [comment]
if you click on your followers, via Tusky / whatever, does it show the full expected list?

**Comment by Fastidious** [comment]
Yup, it does. The numbers are wrong, but the list shows what it should be.

**Comment by tsmethurst** [comment]
Sounds like an AccountStats got populated incorrectly somewhere.

**Comment by tsmethurst** [comment]
Does the issue go away if you restart your instance? Checking if it's a caching problem.

**Comment by Fastidious** [comment]
Will restart as soon as I reach home. Following new accounts is increasing the count.

**Comment by Fastidious** [comment]
Restarting doesn't fix it.

**Comment by tsmethurst** [comment]
Could you do `SELECT * FROM account_stats WHERE id = '01J6QGKXRQCQWZQVDKHNAERCNE';` and see what it outputs? I think the stored account stats got messed up somewhere. Not really a big problem since they'll regenerate after 48 hours or something anyway (so if you wait a while, this problem will solve itself), but I would like to know a) if that's what happened and b) why.

**Comment by Fastidious** [comment]
I am getting this:

```
$ sqlite3
SQLite version 3.37.2 2022-01-06 13:25:41
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> .open sqlite.db
sqlite> SELECT * FROM account_stats WHERE id = '01J6QGKXRQCQWZQVDKHNAERCNE';
Error: in prepare, no such column: id (1)
```

**Comment by tsmethurst** [comment]
oh sorry, try `SELECT * FROM account_stats WHERE account_id = '01J6QGKXRQCQWZQVDKHNAERCNE';`

**Comment by Fastidious** [comment]
That renders:

```
sqlite> SELECT * FROM account_stats WHERE account_id = '01J6QGKXRQCQWZQVDKHNAERCNE';
01J6QGKXRQCQWZQVDKHNAERCNE|2024-07-12 20:14:34.834103+00:00|2|5|0|7|0|2024-07-13 00:52:36+00:00
```

Which are the wrong counts.

**Comment by tsmethurst** [comment]
Alright thanks. Since your "last regenerated" value is "2024-07-12 20:14:34.834103+00:00", if you wait til 8pm tomorrow (ie., 48hrs since the stats were last regenerated), the stats will be regenerated again and should be correct then. Not sure what went wrong before then though. You haven't been doing any manual messing around in the database have you?

**Comment by Fastidious** [comment]
I changed joined date on Monday. Ever since I updated two, or three times without issues. My DB backup, before upgrading to `3.5.3+0.16.1-SNAPSHOT+git-aeb65bc`, has the right values:

```
sqlite> SELECT * FROM account_stats WHERE account_id = '01J6QGKXRQCQWZQVDKHNAERCNE';
01J6QGKXRQCQWZQVDKHNAERCNE|2024-07-07 21:32:54.020818+00:00|13|123|0|357|1|2024-07-08 14:38:48.726605+00:00
```

**Comment by tsmethurst** [comment]
Alright thanks for checking. I'd say sit tight a couple days and see if the stats get back to normal. This is clearly a bug but I'm knee-deep in a bunch of other stuff rn so no time to investigate fully at the moment.

**Comment by Fastidious** [comment]
Statistics just fixed themselves.
